api/zulip/__init__.py

```diff
@@ -290,16 +290,26 @@ def get_user_agent(self):
                 vendor_version=vendor_version,
                 )
 
-    def do_api_query(self, orig_request, url, method="POST", longpolling=False):
-        # type: (Mapping[str, Any], str, str, bool) -> Dict[str, Any]
+    def do_api_query(self, orig_request, url, files=None, method="POST", longpolling=False):
+        # type: (Mapping[str, Any], str, List[typing.IO], str, bool) -> Dict[str, Any]
         request = {}
+        req_files = []
 
         for (key, val) in six.iteritems(orig_request):
             if isinstance(val, str) or isinstance(val, six.text_type):
                 request[key] = val
             else:
                 request[key] = simplejson.dumps(val)
 
+        if isinstance(files, list):
+            for file in files:
+                req_files.append((file.name, file))
+        else:
+            try:
+                req_files.append((files.name, files))
+            except AttributeError:
+                raise TypeError('files must be a file list/single object')
```

Comment:
Probably rather than both accepting single files and lists, I'd just accept lists and have `upload_file` pass a list in as e.g. `[file]`.